kornia/feature/laf.py

```diff
@@ -196,7 +196,7 @@ def make_upright(laf: torch.Tensor, eps: float = 1e-9) -> torch.Tensor:
     scale = det
     # The function is equivalent to doing 2x2 SVD and resetting rotation
     # matrix to an identity: U, S, V = svd(LAF); LAF_upright = U * S.
-    b2a2 = torch.sqrt(laf[..., 0:1, 1:2] ** 2 + laf[..., 0:1, 0:1] ** 2) + eps
+    b2a2 = (torch.sqrt(laf[..., 0:1, 1:2] ** 2 + laf[..., 0:1, 0:1] ** 2) + eps).type_as(laf)
```

Comment:
is it really needed here ? the output of this line is based on `laf` already

Reply:
All the power of 2 functions seem to promote float16 to float32. I'm not sure what the proper way of dealing with this is.

Reply:
Interesting, I’ll research on that

Reply:
@t-vi any suggestion here ?